fcrepo-kernel/src/main/java/org/fcrepo/kernel/observer/FedoraEvent.java

```diff
@@ -59,6 +65,7 @@
      */
     public FedoraEvent(final Event e) {
         checkArgument(e != null, "null cannot support a FedoraEvent!");
+        eventID = pidMinter.mintPid();
```

Comment:
CDI can inject parameters into a constructor for _any_ class, which automatically becomes managed. So a class that needs a `FedoraEvent` could just use the constructor and CDI, behind the scenes, could inject an appropriate identifier minter.